Add more neurio energy sensors

BobbyBleacher opened the pull request:
## Breaking Change:

None known

## Description:

Adds in Solar Gen (active), Solar Gen (total) and Net consumption (used-generated).

## Example entry for `configuration.yaml` (if applicable):
```yaml
Adds sensors:
    - sensor.solar_generation
    - sensor.generation_total
    - sensor.net_consumption
```

BobbyBleacher committed:
More sensors

Adds in Solar Gen (active), Solar Gen (total) and Net consumption (used-generated).

homeassistant commented:
Hi @BobbyBleacher, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#23106).

Once you do that we will be able to review and accept this pull request.

Thanks!

BobbyBleacher committed:
Minor updates

A few small changes to efficiency, error handling.

homeassistant commented:
Hi @BobbyBleacher, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#23106).

Once you do that we will be able to review and accept this pull request.

Thanks!

BobbyBleacher committed:
Code refactoring, bug fixes

Changes:
-Now makes only 1 request to the rest API to update stats. The way it was written originally caused it to call the API for every piece of data it wanted to update. This caused the script to reach the rest API rate limit very quickly. Should now be able to update about every 24s per hour (144/hr rate limit). Set to update at a min of every 30s.
-Included the need to specify the neurio IP address, so solar gen and active power could be updated locally rather than through the rest API (allows for more frequent updates)

BobbyBleacher committed:
Bug Fixes

BobbyBleacher committed:
More bug fixes

BobbyBleacher committed:
Syntax fixes to pass circleci

BobbyBleacher committed:
More fixes for validation testing

BobbyBleacher committed:
Validation Fixed

BobbyBleacher committed:
More edits for validation

BobbyBleacher committed:
Update sensor.py

BobbyBleacher committed:
Update sensor.py

BobbyBleacher committed:
Update sensor.py

BobbyBleacher committed:
Update sensor.py

BobbyBleacher committed:
Update sensor.py

BobbyBleacher committed:
Update sensor.py

BobbyBleacher committed:
Update sensor.py

balloob commented on homeassistant/components/neurio_energy/sensor.py:
```diff
@@ -17,22 +20,28 @@
 
-MIN_TIME_BETWEEN_DAILY_UPDATES = timedelta(seconds=150)
-MIN_TIME_BETWEEN_ACTIVE_UPDATES = timedelta(seconds=10)
+MIN_TIME_BETWEEN_DAILY_UPDATES = timedelta(seconds=30)
```
Please revert this. We don't want to hammer the API

BobbyBleacher replied:
No. We're not hammering the API. Per the documentation that is an acceptable request and no longer generates the multiple request errors as the official code.

balloob commented on homeassistant/components/neurio_energy/sensor.py:
```diff
@@ -54,28 +69,81 @@ def update_active():
+                 )
     # Active power sensor
-    add_entities([NeurioEnergy(data, ACTIVE_NAME, ACTIVE_TYPE, update_active)])
+    add_entities([NeurioEnergy(data,
```
Please revert all the style changes to this file. They make it impossible to see what has changed.

BobbyBleacher replied:
What?

BobbyBleacher replied:
I had to do this to get the code to pass the validation. Otherwise I kept getting errors for the lines being too long.

balloob commented on homeassistant/components/neurio_energy/sensor.py:
```diff
@@ -99,35 +172,94 @@ def active_power(self):
     def get_active_power(self):
         """Return current power value."""
+        header = {'Authorization': 'bearer <token>'}
+        resp = requests.get('http://'+self.sensor_ip+'/current-sample',
```
This needs to be added to the `neurio` package. Home Assistant does not contain protocol specific information.

BobbyBleacher replied:
Not sure what you mean.

MartinHjelmare replied:
Please make a pull request to the neurio library repo and add the new data fetching methods. Once a new version of the library has been released to pypi. A pull request can be made to home assistant which upgrades the used library version and uses the new methods.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#4-communication-with-devices-services

BobbyBleacher replied:
I see. I figured my component would just replace the original. I'll make the PR there.

BobbyBleacher replied:
Isn't this what I did?

MartinHjelmare replied:
I don't think so. Here's the repo of the neurio library:
https://github.com/jordanh/neurio-python

You should submit a PR there to add the necessary new methods. After that is accepted there needs to be a new version released of the neurio library to PyPI. When that is done we can bump the version of the neurio library used in home assistant in another PR.

MartinHjelmare commented:
@BobbyBleacher are you planning to finish this?

balloob commented:
This PR seems to have gone stale. Closing it.

BobbyBleacher commented:
It is finished. What else do you need?

BobbyBleacher commented:
> @BobbyBleacher are you planning to finish this?

It IS finished. Not sure what else you need. It's not stale, I didn't get any notifications from GitHub about your messages. Re-open.

MartinHjelmare commented:
Please address the comments above. If there's something not clear with the commenta, please ask specific questions so we can help answering them.

MartinHjelmare commented:
There's no movement here in the two weeks since I reopened the PR. I'll close now. Please open a new PR when all the comments have been addressed.
